pkg/tests/ossm/smoke_test.go

```diff
@@ -19,50 +19,195 @@ import (
+		t.NewSubTest(fmt.Sprintf("upgrade %s to %s", fromVersion, toVersion)).Run(func(t TestHelper) {
+			t.Logf("This test checks whether SMCP becomes ready after it's upgraded from %s to %s and bookinfo is still working after the upgrade", fromVersion, toVersion)
+
+			t.Cleanup(func() {
+				app.Uninstall(t, app.Bookinfo(ns), app.SleepNoSidecar(ns))
+			})
+
+			t.LogStepf("Upgrade SMCP from %s to %s", fromVersion, toVersion)
+			assertSMCPDeploysAndIsReady(t, toVersion)
+
+			t.LogStep("Check if bookinfo productpage is running through the Proxy after the upgrade")
+			retry.UntilSuccess(t, func(t test.TestHelper) {
+				oc.Exec(t,
+					pod.MatchingSelector("app=sleep", ns), "sleep",
+					"curl -sI http://productpage:9080",
+					assert.OutputContains(
+						"HTTP/1.1 200 OK",
+						"ProductPage returns 200 OK",
+						"ProductPage didn't return 200 OK"),
+					assert.OutputContains(
+						"server: istio-envoy",
+						"HTTP header 'server: istio-envoy' is present in the response",
+						"HTTP header 'server: istio-envoy' is missing from the response"),
+					assert.OutputContains(
+						"x-envoy-decorator-operation",
+						"HTTP header 'x-envoy-decorator-operation' is present in the response",
+						"HTTP header 'x-envoy-decorator-operation' is missing from the response"))
+			})
+
+			t.LogStep("Delete Bookinfo pods to validate proxy is still working after recreation and upgrade")
+			oc.RestartAllPodsAndWaitReady(t, ns)
+			retry.UntilSuccess(t, func(t test.TestHelper) {
+				oc.Exec(t,
+					pod.MatchingSelector("app=sleep", ns), "sleep",
+					"curl -sI http://productpage:9080",
+					assert.OutputContains(
+						"HTTP/1.1 200 OK",
+						"ProductPage returns 200 OK",
+						"ProductPage didn't return 200 OK"),
+					assert.OutputContains(
+						"server: istio-envoy",
+						"HTTP header 'server: istio-envoy' is present in the response",
+						"HTTP header 'server: istio-envoy' is missing from the response"),
+					assert.OutputContains(
+						"x-envoy-decorator-operation",
+						"HTTP header 'x-envoy-decorator-operation' is present in the response",
+						"HTTP header 'x-envoy-decorator-operation' is missing from the response"))
+			})
+		})
+
+		t.NewSubTest(fmt.Sprintf("delete smcp %s", toVersion)).Run(func(t TestHelper) {
+			t.Logf("This test checks whether SMCP %s deletion delete all the resources", env.GetSMCPVersion())
+
+			t.LogStep("Delete SMCP and verify if this deletes all resources")
+			assertUninstallDeletesAllResources(t, env.GetSMCPVersion())
+		})
 	})
 }
 
+func validateStartUpProxyTime(t TestHelper, ratingYaml string) {
+	proxyStartTime, proxyLastTransitionTime := ExtractProxyTimes(t, ratingYaml)
+	startupTime := proxyLastTransitionTime.Sub(proxyStartTime)
+	t.Logf("Proxy startup time: %s", startupTime.String())
+	if startupTime > 10*time.Second {
+		t.Fatalf("Proxy startup time is too long: %s", startupTime.String())
+	}
+}
+
+func ExtractProxyTimes(t TestHelper, yamlString string) (time.Time, time.Time) {
```

Comment:
This would be much shorter and simpler if you changed `oc.GetYaml()` to accept `jsonPath string` instead of `checks ...common.CheckFunc` to directly extract the two timestamps. As far as I can see, we never actually pass any check functions, so we can get rid of this parameter.

Reply:
I need to open a new Jira for this, because cluster wide test use, so I need to make change in more than one test at the same time. https://issues.redhat.com/browse/OSSM-4032

Reply:
Actually, if you pass in jsonpath, you get back JSON, not YAML, so you could simply create a new function: `oc.GetJSON(..., jsonPath string)`. So, you don't have to change any existing code.